tools/gen_addon_readme.py

```diff
@@ -117,29 +118,61 @@ def make_weblate_badge(repo_name, branch, addon_name):
     )
 
 
-def make_repo_badge(repo_name, branch, addon_name):
+def make_repo_badge(org_name, repo_name, branch, addon_name):
     badge_repo_name = repo_name.replace('-', '--')
     return (
-        'https://img.shields.io/badge/github-OCA%2F{badge_repo_name}'
+        'https://img.shields.io/badge/github-{org_name}%2F{badge_repo_name}'
         '-lightgray.png?logo=github'.format(**locals()),
-        'https://github.com/OCA/{repo_name}/tree/'
+        'https://github.com/{org_name}/{repo_name}/tree/'
         '{branch}/{addon_name}'.format(**locals()),
-        'OCA/{repo_name}'.format(**locals()),
+        '{org_name}/{repo_name}'.format(**locals()),
     )
 
 
-def gen_one_addon_readme(repo_name, branch, addon_name, addon_dir, manifest):
+def generate_fragment(org_name, repo_name, branch, addon_name, file):
+    fragment_lines = file.readlines()
+    if not fragment_lines:
+        return False
+
+    # Replace relative path by absolute path for figures
+    module_url = "https://raw.githubusercontent.com/{org_name}/{repo_name}"\
+        "/{branch}/{addon_name}/".format(**locals())
+    for index, fragment_line in enumerate(fragment_lines):
+        if ('.. figure::' in fragment_line or
+                '.. image::' in fragment_line):
+            path = fragment_line.strip()\
+                .replace('\n', '')\
+                .replace('.. figure:: ', '')\
+                .replace('.. image:: ', '')
+            if path.startswith('http'):
+                # It is already an absolute path
+                continue
+            else:
+                # remove '../' if exists that make the fragment working
+                # on github interface, in the 'readme' subfolder
+                relative_path = path.replace('../', '')
+                fragment_lines[index] = fragment_line.replace(
+                    path, urljoin(module_url, relative_path))
+    fragment = ''.join(fragment_lines)
+
+    # ensure that there is a new empty line at the end of the fragment
+    if fragment[-1] != '\n':
+        fragment += '\n'
+    return fragment
+
+
+def gen_one_addon_readme(
+        org_name, repo_name, branch, addon_name, addon_dir, manifest):
     fragments = {}
     for fragment_name in FRAGMENTS:
         fragment_filename = os.path.join(
             addon_dir, FRAGMENTS_DIR, fragment_name + '.rst',
         )
         if os.path.exists(fragment_filename):
             with io.open(fragment_filename, 'rU', encoding='utf8') as f:
-                fragment = f.read()
+                fragment = generate_fragment(
+                    org_name, repo_name, branch, addon_name, f)
                 if fragment:
-                    if fragment[-1] != '\n':
-                        fragment += '\n'
                     fragments[fragment_name] = fragment
     runbot_id = get_runbot_ids().get(repo_name)
```

Comment:
I suggest looking up runbot_id only if org is OCA?

Reply:
indeed, done.